src/body.rs

```diff
@@ -24,11 +24,38 @@ impl<T: Payload> LiftBody<T> {
     }
 }
 
-impl<T: Payload> BufStream for LiftBody<T> {
+impl<T: Payload> HttpBody for LiftBody<T> {
     type Item = <T as Payload>::Data;
     type Error = <T as Payload>::Error;
 
     fn poll_buf(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
         self.inner.poll_data()
     }
+
+    fn poll_trailers(&mut self) -> Poll<Option<hyper::HeaderMap>, Self::Error> {
+        self.inner.poll_trailers()
+    }
+
+    fn is_end_stream(&self) -> bool {
+        self.inner.is_end_stream()
+    }
+}
+
+impl<T> Payload for LiftBody<T>
+    where T: HttpBody + Send + 'static,
+          T::Item: Send,
+          T::Error: Into<Box<std::error::Error + Send + Sync>>,
```

Comment:
minor nit: can we move the `Box<...>` to a `pub(crate) type Error = Box<...>`, this is just a pattern we have been using throughout tower.